spec/queue.js

```diff
@@ -19,4 +19,87 @@ describe('Queue', () => {
         .to.alter(() => myQueue.length(), { from: 0, to: 1 })
     })
   })
+
+  context('dequeue()', () => {
+    it('returns and removes the front element in the queue or null if the queue is empty.', () => {
```

Comment:
This is describing two tests (and you have the test for null return when queue is empty below).  Consider changing this to describe the assertion under test.